src/net.c

```diff
@@ -753,6 +759,198 @@ static int hyper_set_interface_mtu(struct rtnl_handle *rth,
 				IFLA_MTU);
 }
 
+static char* hyper_ifreq_to_mac_addr(struct ifreq ifr)
+{
+	char *mac_addr;
+
+	if (asprintf(&mac_addr,
+		     "%02x:%02x:%02x:%02x:%02x:%02x",
+		     (uint8_t)ifr.ifr_hwaddr.sa_data[0],
+		     (uint8_t)ifr.ifr_hwaddr.sa_data[1],
+		     (uint8_t)ifr.ifr_hwaddr.sa_data[2],
+		     (uint8_t)ifr.ifr_hwaddr.sa_data[3],
+		     (uint8_t)ifr.ifr_hwaddr.sa_data[4],
+		     (uint8_t)ifr.ifr_hwaddr.sa_data[5]) == -1) {
+		fprintf(stderr, "failed to allocate mac_addr string\n");
+		return NULL;
+	}
+
+	return mac_addr;
+} 
+
+/*!
+ * Check hardware address related to the provided network interface name
+ * matches the expected hardware address. It expects EUI-48 MAC addresses.
+ *
+ * \param device network interface name.
+ * \param mac_addr hardware address (EUI-48) to match with \p device one.
+ *
+ * \note In case the function succeeds, it returns 0. In case the function
+ * fails, it returns -1.
+ */
+static int hyper_check_device_match_mac_addr(const char *device,
+					     const char *mac_addr)
+{
+	struct ifreq ifr;
+	int sock = -1;
+	char *tmp_mac_addr = NULL;
+	int ret = -1;
+
+	if (!device && !strcmp(device, "")) {
+		fprintf(stderr, "invalid device\n");
+		goto err;
+	}
+
+	if (!mac_addr && !strcmp(mac_addr, "")) {
+		fprintf(stderr, "invalid mac_addr\n");
+		goto err;
+	}
+
+	sock = socket(PF_INET, SOCK_DGRAM, 0);
+	if (sock == -1) {
+		fprintf(stderr, "failed to get socket handle: %s\n",
+			strerror(errno));
+		goto err;
+	}
+
+	if (!strncpy(ifr.ifr_name, device, strlen(device) + 1)) {
+		fprintf(stderr, "strncpy failed to copy interface"
+			" name: %s\n", strerror(errno));
+		goto err;
+	}
+
+	if (ioctl(sock, SIOCGIFHWADDR, &ifr) == -1) {
+		fprintf(stderr, "ioctl SIOCGIFHWADDR failed: %s\n",
+			strerror(errno));
+		goto err;
+	}
+
+	tmp_mac_addr = hyper_ifreq_to_mac_addr(ifr);
+	if (!tmp_mac_addr) {
+		fprintf(stderr, "tmp_mac_addr is NULL\n");
+		goto err;
+	}
+
+	if (!strcasecmp(mac_addr, tmp_mac_addr)) {
+		fprintf(stderr, "device mac address found %s does not match"
+			" with the expected mac address %s\n",
+			tmp_mac_addr, mac_addr);
+		goto err;
+	}
+
+	ret = 0;
+
+err:
+	free(tmp_mac_addr);
+	close(sock);
+	return ret;
+}
+
+/*!
+ * Find network interface list, and retrieve the name of the one
+ * matching the EUI-48 hardware address provided.
+ *
+ * \param mac_addr hardware address (EUI-48) to match.
+ * \param[out] device name of the network interface matching \p mac_addr.
+ *
+ * \note In case the function succeeds, it returns 0 and \p device is
+ * filled with the right network interface name. In case the function
+ * fails, it returns -1 and \p device will point to \c NULL.
+ */
+static int hyper_get_iface_name_from_mac_addr(const char *mac_addr,
```

Comment:
Well I have reversed the arguments but there is a slight difference between those functions. First, we are not checking device the same way. And second, the strcasecmp() does not result in the same thing.
Moreover, I don't think it would be very efficient to have a function opening the socket and closing called at every iteration of the loop.
I think the best I can do is to factorize what is inside the loop. Do you agree ?